pcsx2/CDVD/CDVDdiscReader.h

```diff
@@ -0,0 +1,111 @@
+/*  PCSX2 - PS2 Emulator for PCs
+ *  Copyright (C) 2002-2020  PCSX2 Dev Team
+ *
+ *  PCSX2 is free software: you can redistribute it and/or modify it under the terms
+ *  of the GNU Lesser General Public License as published by the Free Software Found-
+ *  ation, either version 3 of the License, or (at your option) any later version.
+ *
+ *  PCSX2 is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ *  PURPOSE.  See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with PCSX2.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef __CDVD_DISC_READER_H__
+#define __CDVD_DISC_READER_H__
+
+#include "IopCommon.h"
+
+#if defined(_WIN32)
+#define _WIN32_WINNT 0x0600
+#define NOMINMAX
+#include <windows.h>
+#endif
+
+#include <mutex>
+#include <array>
+
+struct track
+{
+	u32 start_lba;
+	u8 type;
+};
+
+extern u8 strack;
+extern u8 etrack;
+extern track tracks[100];
+
+extern int curDiskType;
+extern int curTrayStatus;
+
+struct toc_entry
+{
+	u32 lba;
+	u8 track;
+	u8 adr : 4;
+	u8 control : 4;
+};
+
+class IOCtlSrc
```

Comment:
I think the definitions for the member functions of this class won't exist on OSes other than Windows and Linux.